### ISSUE TITLE
[Question] Authentication flow

### ISSUE THREAD

**Issue by GioF71** [tehkillerbee/mopidy-tidal#176]
Hello, with referent to [this issue](https://github.com/tamland/python-tidal/issues/277), can we use the regular oauth2 auth flow and get HI_RES_LOSSLESS quality on mopidy-tidal as well?

**Comment by tehkillerbee** [comment]
Yes, I believe that should work as well but I have yet to test it.

**Comment by fmarzocca** [comment]
What's the point not using pkce?

**Comment by GioF71** [comment]
> What's the point not using pkce?

AFAIK currently you cannot get new authentication tokens, at least not with the known workflow. I personally get an error when I try that.

**Comment by GioF71** [comment]
> What's the point not using pkce?

and also it would be better if we can avoid the complexity of the process. The regular flow is simpler

**Comment by tehkillerbee** [comment]
> > What's the point not using pkce?
> 
> AFAIK currently you cannot get new authentication tokens, at least not with the known workflow. I personally get an error when I try that.

Can you elaborate the error? PKCE still works when using python-tidal directly.

**Comment by GioF71** [comment]
> > > What's the point not using pkce?
> > 
> > 
> > AFAIK currently you cannot get new authentication tokens, at least not with the known workflow. I personally get an error when I try that.
> 
> Can you elaborate the error? PKCE still works when using python-tidal directly.

with your existing token, yes, it also works for me. Have you tried to get a new one? Here's when the error happens for me.

**Comment by tehkillerbee** [comment]
> > > > What's the point not using pkce?
> > > 
> > > 
> > > AFAIK currently you cannot get new authentication tokens, at least not with the known workflow. I personally get an error when I try that.
> > 
> > 
> > Can you elaborate the error? PKCE still works when using python-tidal directly.
> 
> with your existing token, yes, it also works for me. But have you tried to get a new one? Here's when the error happens for me.

Just tried (using the pkce_example in python-tidal) creating a new token. It still appears to work as expected. Are you trying to authenticate through mopidy-tidal?

**Comment by fmarzocca** [comment]
> Hello, with referent to [this issue](https://github.com/tamland/python-tidal/issues/277), can we use the regular oauth2 auth flow and get HI_RES_LOSSLESS quality on mopidy-tidal as well?

BTW, I am using OAUTH (due to limitation of my audio decoder) and I am NOT getting any HI_RES_LOSSLESS.

**Comment by GioF71** [comment]
> > > > > What's the point not using pkce?
> > > > 
> > > > 
> > > > AFAIK currently you cannot get new authentication tokens, at least not with the known workflow. I personally get an error when I try that.
> > > 
> > > 
> > > Can you elaborate the error? PKCE still works when using python-tidal directly.
> > 
> > 
> > with your existing token, yes, it also works for me. But have you tried to get a new one? Here's when the error happens for me.
> 
> Just tried (using the pkce_example in python-tidal) creating a new token. It still appears to work as expected. Are you trying to authenticate through mopidy-tidal?

Alright, in fact today it works again. yesterday I was getting an error exactly when I had to enter the username on the form. Ok, so this problem is apparently not here anymore.
